packages/combobox/src/Combobox.tsx

```diff
@@ -0,0 +1,923 @@
+        if (isComponentType(child, 'ComboboxOption')) {
+          const { value, displayName } = getNameAndValue(child.props);
+
+          if (isOptionVisible(value)) {
+            const { className, glyph } = child.props;
+            const index = allOptions.findIndex(opt => opt.value === value);
+
+            const isFocused = focusedOption === value;
+            const isSelected = isMultiselect(selection)
+              ? selection.includes(value)
+              : selection === value;
+
+            const setSelected = () => {
+              setFocusedOption(value);
+              updateSelection(value);
+            };
+
+            const optionRef = getOptionRef(value);
+
+            return (
+              <InternalComboboxOption
+                value={value}
+                displayName={displayName}
+                isFocused={isFocused}
+                isSelected={isSelected}
+                setSelected={setSelected}
+                glyph={glyph}
+                className={className}
+                index={index}
+                ref={optionRef}
+              />
+            );
+          }
+        } else if (isComponentType(child, 'ComboboxGroup')) {
+          const nestedChildren = renderInternalOptions(child.props.children);
+
+          if (nestedChildren && nestedChildren?.length > 0) {
+            return (
+              <InternalComboboxGroup
+                label={child.props.label}
+                className={child.props.className}
+              >
+                {renderInternalOptions(nestedChildren)}
+              </InternalComboboxGroup>
+            );
+          }
+        }
+      });
+    },
+    [
+      allOptions,
+      focusedOption,
+      getOptionRef,
+      isMultiselect,
+      isOptionVisible,
+      selection,
+      updateSelection,
+    ],
+  );
+
+  const renderedOptions = useMemo(() => renderInternalOptions(children), [
+    children,
+    renderInternalOptions,
+  ]);
+
+  const renderedChips = useMemo(() => {
+    if (isMultiselect(selection)) {
+      return selection.filter(isValueValid).map(value => {
+        const displayName = getDisplayNameForValue(value);
+
+        const onRemove = () => {
+          updateSelection(value);
+        };
+
+        const isFocused = focusedChip === value;
+        const chipRef = getChipRef(value);
+
+        return (
+          <Chip
+            key={value}
+            displayName={displayName}
+            onRemove={onRemove}
+            isFocused={isFocused}
+            ref={chipRef}
+          />
+        );
+      });
+    }
+  }, [
+    isMultiselect,
+    selection,
+    isValueValid,
+    getDisplayNameForValue,
+    focusedChip,
+    getChipRef,
+    updateSelection,
+  ]);
+
+  const renderedInputIcons = useMemo(() => {
+    return (
+      <>
+        {clearable && doesSelectionExist && (
+          <IconButton
+            ref={clearButtonRef}
+            onClick={() => updateSelection(null)}
```

Comment:
This does not appear to cause the example to re-validate your selection.

Repro: 
- Select "Dragonfruit", causing an error to appear
- Clear your current selections
- The error remains until you make a new selection